book_programs/chapter_2/bertrands_paradox.py

```diff
@@ -0,0 +1,191 @@
+#   Usage: python3 bertrands_paradox.py <NCHORDS_TO_PLOT> <nchords> 
+#   NCHORDS_TO_PLOT:  The number of chords to plot
+#   Don't plot more than 1000 number of chords because they overlap too much
+#   and obscure the point we're trying to make.
+#   nchords:  The sample size of chords to do the statistics
+
+# Example Usage: python3 bertrands_paradox.py 1000 10000
+
+import numpy as np
+import sys
+import matplotlib.pyplot as plt
+from matplotlib.patches import Circle
+from matplotlib.lines import Line2D
+
+args = sys.argv
+
+#####################
+# User Input Handling
+#####################
+bad_input = False
+
+# Less than 2 arguments
+if len(args) < 2:
+    bad_input = True
+    print("ERROR: Insufficient arguments")

```

Comment:
If there's only 1 argument passed to the script, an error is thrown. The first argument is always the name of the script. 
`python3 bertrands_paradox.py 10` would have args `["bertrands_paradox.py", "10"]` so there would be 2 args but `NCHORDS_TO_PLOT` would not have a value passed to it causing the error
I missed this for buffons_needle but I fixed it

Reply:
"bertrands_paradox.py" is args[0], right ?

Reply:
yes